Source/Particles/Deposition/TemperatureDeposition.H

```diff
@@ -54,61 +56,81 @@ void varianceDepositionSubKernel(
-        &varx_arr(ixv[0], ixv[1], ixv[2], 0),
-        wqx_var*wx_old*vxbar_diff*vxbar_diff/wxsum);
-    amrex::Gpu::Atomic::AddNoRet(
-        &vary_arr(iyv[0], iyv[1], iyv[2], 0),
-        wqy_var*wy_old*vybar_diff*vybar_diff/wysum);
-    amrex::Gpu::Atomic::AddNoRet(
-        &varz_arr(izv[0], izv[1], izv[2], 0),
-        wqz_var*wz_old*vzbar_diff*vzbar_diff/wzsum);
+    const amrex::Real vxr = static_cast<amrex::Real>(vx);
+    const amrex::Real vyr = static_cast<amrex::Real>(vy);
+    const amrex::Real vzr = static_cast<amrex::Real>(vz);
+
+    if (type == TemperatureDepositionType::SINGLE_PASS || pass == TemperatureDepositionPass::FIRST)
+    {
+        // Update sample count
+        amrex::Gpu::Atomic::AddNoRet(
+            &nx_arr(ixv[0], ixv[1], ixv[2], 0), 1);
+        amrex::Gpu::Atomic::AddNoRet(
+            &ny_arr(iyv[0], iyv[1], iyv[2], 0), 1);
+        amrex::Gpu::Atomic::AddNoRet(
+            &nz_arr(izv[0], izv[1], izv[2], 0), 1);
+
+        // Update w_sum
+        amrex::Gpu::Atomic::AddNoRet(
+            &wx_arr(ixv[0], ixv[1], ixv[2], 0), wqx_var);
+        amrex::Gpu::Atomic::AddNoRet(
+            &wy_arr(iyv[0], iyv[1], iyv[2], 0), wqy_var);
+        amrex::Gpu::Atomic::AddNoRet(
+            &wz_arr(izv[0], izv[1], izv[2], 0), wqz_var);
+    }
+
+    if (type == TemperatureDepositionType::SINGLE_PASS || pass == TemperatureDepositionPass::SECOND)
+    {
+        // For a single pass run through, the vbar arrays will be zeroed, so this should work
+        // in either case. Will be non-demeaned in 1 pass
+        amrex::Real vxb = 0.;
+        amrex::Real vyb = 0.;
+        amrex::Real vzb = 0.;
+
+        // Normalization of accumulation arrays are not done until after all deposition, so nomralize
+        // here for double pass algorithm
+        if (type == TemperatureDepositionType::DOUBLE_PASS)
+        {
+            if (nx_arr(ixv[0], ixv[1], ixv[2], 0) > 0) {
+                vxb = vxbar_arr(ixv[0], ixv[1], ixv[2], 0)/wx_arr(ixv[0], ixv[1], ixv[2], 0);
+            }
+            if (ny_arr(iyv[0], iyv[1], iyv[2], 0) > 0) {
+                vyb = vybar_arr(iyv[0], iyv[1], iyv[2], 0)/wy_arr(iyv[0], iyv[1], iyv[2], 0);
+            }
+            if (nz_arr(izv[0], izv[1], izv[2], 0) > 0) {
+                vzb = vzbar_arr(izv[0], izv[1], izv[2], 0)/wz_arr(izv[0], izv[1], izv[2], 0);
+            }
+
+        }
+
+        const amrex::Real vxd = vxr - vxb;
+        const amrex::Real vyd = vyr - vyb;
+        const amrex::Real vzd = vzr - vzb;
+
+            // Update wv2_sum
+        amrex::Gpu::Atomic::AddNoRet(
+            &w2x_arr(ixv[0], ixv[1], ixv[2], 0), wqx_var*vxd*vxd);
+        amrex::Gpu::Atomic::AddNoRet(
+            &w2y_arr(iyv[0], iyv[1], iyv[2], 0), wqy_var*vyd*vyd);
+        amrex::Gpu::Atomic::AddNoRet(
+            &w2z_arr(izv[0], izv[1], izv[2], 0), wqz_var*vzd*vzd);
+    }
+
+    if (type == TemperatureDepositionType::SINGLE_PASS || pass == TemperatureDepositionPass::FIRST)
```

Comment:
Is there a reason that this `if` block is not included in the one above at line 68?